engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -35,83 +38,24 @@
 
-    public Uni<Endpoint> createEndpoint(Endpoint endpoint, boolean useStatelessSession) {
-        return commonStateSessionFactory.withSession(useStatelessSession, session -> {
-            return session.persist(endpoint)
-                    .onItem().call(session::flush)
-                    .onItem().call(() -> {
-                        // If the endpoint properties are null, they won't be persisted.
-                        if (endpoint.getProperties() != null) {
-                            /*
-                             * As weird as it seems, we need the following line because the Endpoint instance was
-                             * deserialized from JSON and that JSON did not contain any information about the
-                             * @OneToOne relation from EndpointProperties to Endpoint.
-                             */
-                            endpoint.getProperties().setEndpoint(endpoint);
-                            switch (endpoint.getType()) {
-                                case CAMEL:
-                                case WEBHOOK:
-                                case EMAIL_SUBSCRIPTION:
-                                    return session.persist(endpoint.getProperties())
-                                            .onItem().call(session::flush);
-                                default:
-                                    // Do nothing.
-                                    break;
-                            }
-                        }
+    public Uni<Endpoint> getOrCreateEmailSubscriptionEndpoint(String accountId) {
+        String query = "FROM Endpoint WHERE accountId = :accountId AND type = :endpointType";
+        return commonStateSessionFactory.withSession(true, session -> {
+            return session.createQuery(query, Endpoint.class)
+                    .setParameter("accountId", accountId)
+                    .setParameter("endpointType", EMAIL_SUBSCRIPTION)
+                    .getResultList()
+                    .call(this::loadProperties)
+                    .onItem().transformToUni(emailEndpoints -> {
                         /*
-                         * If this line is reached, it means the endpoint properties are null or we don't support
-                         * persisting properties for the endpoint type. We still have to return something.
+                         * The purpose of the following code is to find or create an EMAIL_SUBSCRIPTION endpoint with
+                         * empty properties. This endpoint is used to aggregate and store in the DB the email actions
+                         * outcome, which will be used later by the event log. The recipients of the current email
+                         * action have already been resolved before this step, possibly from multiple endpoints and
+                         * recipients settings. The properties created below have no impact on the resolution of the
+                         * action recipients.
```

Comment:
@josejulio Is that clear enough?

Reply:
The description seems good to me, but I'm unsure if the location is the best. 

The name of the method is the same as in the `EndpointResources`, and we use it there to create an EmailSubscription with parameters - it could be confusing to the reader.
So maybe moving the comment to where we actualy call this function or renaming the name of the method (getOrCreateDefaultEmailSubscription?)

Reply:
I renamed the method as you suggested and moved the comment to the method signature.